src/NuGetGallery.Services/PackageManagement/PackageOwnershipManagementService.cs

```
@@ -7,29 +7,54 @@
             IEntitiesContext entitiesContext,
             IPackageService packageService,
             IReservedNamespaceService reservedNamespaceService,
             IPackageOwnerRequestService packageOwnerRequestService,
-            IAuditingService auditingService)
+            IAuditingService auditingService,
+            IUrlHelper urlHelper,
+            IAppConfiguration appConfiguration,
+            IMessageService messageService)
         {
             _entitiesContext = entitiesContext ?? throw new ArgumentNullException(nameof(entitiesContext));
             _packageService = packageService ?? throw new ArgumentNullException(nameof(packageService));
             _reservedNamespaceService = reservedNamespaceService ?? throw new ArgumentNullException(nameof(reservedNamespaceService));
             _packageOwnerRequestService = packageOwnerRequestService ?? throw new ArgumentNullException(nameof(packageOwnerRequestService));
             _auditingService = auditingService ?? throw new ArgumentNullException(nameof(auditingService));
+            _urlHelper = urlHelper ?? throw new ArgumentNullException(nameof(urlHelper));
+            _appConfiguration = appConfiguration ?? throw new ArgumentNullException(nameof(appConfiguration));
+            _messageService = messageService ?? throw new ArgumentNullException(nameof(messageService));
+        }
+
+        public async Task AddPackageOwnerWithMessagesAsync(PackageRegistration packageRegistration, User user)
+        {
+            await AddPackageOwnerAsync(packageRegistration, user, commitChanges: true);
+
+            var packageUrl = _urlHelper.Package(packageRegistration.Id, version: null, relativeUrl: false);
+            foreach (var owner in packageRegistration.Owners)
+            {
+                var emailMessage = new PackageOwnerAddedMessage(_appConfiguration, owner, user, packageRegistration, packageUrl);
+                await _messageService.SendMessageAsync(emailMessage);
```

I know this is existing code, but do you think it'd be worthwhile to ignore exceptions thrown by `SendMessageAsync` until after all messages have been sent?

---

Done.